verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -155,26 +192,56 @@ void IndexingFactsTreeExtractor::ExtractModuleEnd(
 
   if (module_name != nullptr) {
     const Anchor module_end_anchor(*module_name, context_.base);
-    facts_tree_context_.back()->Value().AppendAnchor(module_end_anchor);
+    facts_tree_context_.top().Value().AppendAnchor(module_end_anchor);
   }
-}
+};
 
 void IndexingFactsTreeExtractor::ExtractModuleInstantiation(
     const verible::SyntaxTreeNode& node) {
+  std::vector<verible::TreeSearchMatch> gate_instances =
+      GetListOfGateInstanceFromDataDeclaration(node);
+
   const verible::TokenInfo& type =
       GetTypeTokenInfoFromModuleInstantiation(node);
   const Anchor type_anchor(type, context_.base);
 
-  const verible::TokenInfo& variable_name =
-      GetModuleInstanceNameTokenInfoFromDataDeclaration(node);
-  const Anchor variable_name_anchor(variable_name, context_.base);
+  for (const verible::TreeSearchMatch& instance : gate_instances) {
+    IndexingNodeData indexing_node_data(IndexingFactType::kModuleInstance);
+
+    const verible::TokenInfo& variable_name =
+        GetModuleInstanceNameTokenInfoFromGateInstance(*instance.match);
+    const Anchor variable_name_anchor(variable_name, context_.base);
+    indexing_node_data.AppendAnchor(type_anchor);
+    indexing_node_data.AppendAnchor(variable_name_anchor);
+
+    std::vector<verible::TreeSearchMatch> port_names =
+        FindAllUnqualifiedIds(*instance.match);
 
-  IndexingNodeData indexing_node_data(IndexingFactType::kModuleInstance);
-  indexing_node_data.AppendAnchor(type_anchor);
-  indexing_node_data.AppendAnchor(variable_name_anchor);
+    for (const verible::TreeSearchMatch& port : port_names) {
+      const verible::SyntaxTreeLeaf* leaf = GetIdentifier(*port.match);
+      const Anchor port_name_anchor(leaf->get(), context_.base);
 
-  IndexingFactNode module_instance(indexing_node_data);
-  facts_tree_context_.back()->NewChild(module_instance);
+      indexing_node_data.AppendAnchor(port_name_anchor);
+    }
+
+    IndexingFactNode module_instance(indexing_node_data);
+    facts_tree_context_.top().NewChild(module_instance);
+  }
+}
+
+void IndexingFactsTreeExtractor::ExtractNetDeclaration(
```

Comment:
Should this be renamed `ExtractNetDeclarations` (plural)?  Review the other function names too.

Reply:
The functions are intended to extract single facts not many facts at a time and in cases like `wire a, b' these are extracted separately.